### PR TITLE
ECLBASE now behaves separately from JOBNAME

### LINKED ISSUE #5380: Unavoidable warning
**Describe the bug**
When running  `ert test_run user_config.ert` in` test_data/snake_oil_structure/ert` one gets the warning:
  
`/.../ert/src/ert/_c_wrappers/enkf/ert_config.py:327: ConfigWarning: Can not have both JOBNAME and ECLBASE keywords. ECLBASE ignored, using JOBNAME with value `SNAKE_OIL_STRUCTURE_%d` instead`

Then, if in good faith, the line with ECLBASE is removed, one gets the error:
`user_config.ert:93:1:8:When using SUMMARY keyword, the config must also specify ECLBASE
`

(the test_run also fails, probably for other reasons)

**Expected behaviour**
Actionable warning from ert

**Environment**
 - OS: Ubuntu 22.04
 - ERT/Komodo release: 5.0.x
 - Python version: 3.10
 - Remote/HPC execution involved: no

### PR DESCRIPTION
**Pull request by eivindjahren**
**Issue**
Resolves #5380 

Description of behavior:

* Before, if both ECLBASE and JOBNAME were given, then ECLBASE would be used in
all cases, but now JOBNAME is used for the name of the job in the queue.

* If only one of ECLBASE and JOBNAME is given, there is no change

* If neither are given, then ECLBASE now defaults to ECLBASE<IENS>
 rather than JOB<IENS>.

**Approach**
Changes the meaning of setting both ECLBASE and JOBNAME so that it is consistent with the behavior of ERT prior to 2018. Note that there is some weird behavior left where ECLBASE will default to JOBNAME if it is supplied. This is in order to preserve backwards compatibility.

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Updated documentation
- [x] Ensured new behaviour is tested

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

**Comment by oyvindeide** [comment]
Can we have `JOBNAME` default to the config file name? Because now it becomes required again?

**Comment by eivindjahren** [comment]
@oyvindeide Still not required. Defaults to "job%d".

File: test-data/snake_oil_field/snake_oil_surface.ert [hunk 1]
```diff
@@ -9,6 +9,7 @@ RANDOM_SEED 3593114179000630026631423308983283277868
 
 RUNPATH <STORAGE>/runpath/realization-<IENS>/iter-<ITER>
 ENSPATH <STORAGE>/ensemble
+JOBNAME SNAKE_OIL_<IENS>
```
Comment (oyvindeide) [inline test-data/snake_oil_field/snake_oil_surface.ert:12]: Is this needed then?

**Comment by eivindjahren** [comment]
@oyvindeide We can do that! Surprisingly, unfortunately its a bit more complicated than it should be so I suggest separate PR.

File: test-data/snake_oil_field/snake_oil_surface.ert (hunk 1 above)
Comment (eivindjahren) [reply test-data/snake_oil_field/snake_oil_surface.ert:12]: It's not necessarily needed, but it is recommended by the docs so I thought I would add it. Without this, running on the queue creates a bunch of jobs with the same name.

**Pull request merged**